code/modules/admin/verbs/adminweather.dm

```diff
@@ -0,0 +1,78 @@
+ADMIN_VERB(run_weather, R_ADMIN|R_FUN, "Run Weather", "Triggers specific weather on the z-level you choose.", ADMIN_CATEGORY_WEATHER)
+
+	var/static/list/weather_choices = list()
+	if(!length(weather_choices))
+		for(var/datum/weather/weather_type as anything in subtypesof(/datum/weather))
+			weather_choices[initial(weather_type.type)] = weather_type
+
+	var/datum/weather/weather_choice = tgui_input_list(user, "Choose a weather to run", "Weather", weather_choices)
+	if(!weather_choice)
+		return
+	weather_choice = weather_choices[weather_choice]
+
+	var/turf/current_turf = get_turf(user.mob)
+	var/z_level = tgui_input_number(user, "Z-Level to target", "Z-Level", min_value = 1, max_value = world.maxz, default = current_turf?.z)
+	if(!isnum(z_level))
+		return
+
+	var/static/list/custom_options = list("Default", "Custom", "Cancel")
+	var/custom_choice = tgui_alert(user, "How would you like to run the weather settings?", "Custom Weather", custom_options)
+	switch(custom_choice)
+		if("Default")
+			SSweather.run_weather(weather_choice, z_level) // default settings
+			return
+		if("Cancel")
+			return
+		//if("Custom") venture on lil fella
+
+	var/static/list/area_choices = list()
+	if(!length(area_choices))
+		for(var/area/area_type as anything in typesof(/area))
+			area_choices[initial(area_type.type)] = area_type
+
+	var/area/area_choice = tgui_input_list(user, "Select an area for weather to target", "Target Area", area_choices)
+	if(!area_choice)
+		return
+	area_choice = area_choices[area_choice]
+
+	var/weather_bitflags = input_bitfield(
+		user,
+		"Weather flags - Select the flags for your weather event",
+		"weather_flags",
+		weather_choice::weather_flags,
+	)
+
+	var/datum/reagent/reagent_choice
+	if(ispath(weather_choice, /datum/weather/rain_storm) && (weather_bitflags & (WEATHER_TURFS|WEATHER_MOBS)))
```

Comment:
why the explicit typecheck?

Reply:
So I allowed something like thunder to be used by ANY weather subtypes because I can see it being used for ashstorms, sandstorms, etc. In fact one of the other maintainers suggested thunder with ashstroms on lavaland.

The reagent stuff was really only meant to be used by rain. None of the other weather types are greyscaled like rain to inherit reagent color for their sprites.

Could I disable this check to allow reagents to be used for other weather events? Sure, but it just doesn't make sense. Acid snow? (but the snow is white) A holywater sandstorm?

So I just restricted it to the rain weather subtypes.

Reply:
it seems a bit strange conceptually because rain isn't the only weather effect that could in theory have reagents (it's ugly w that setup proc if we're implying generic behavior it should be generic)

I think if you wanted bespoke setup it should happen in an overridable proc on the datum itself (though you would need to change the setup process some).